### PR TITLE
feat(orchestration): allow to reset temporal workflows

### PR DESCRIPTION
**Pull request by gfyrag**
Global note : The history endpoints will return only last run.

### CODE REVIEW

File: ee/orchestration/internal/storage/migrations.go [hunk 1]
```diff
@@ -7,18 +7,10 @@ import (
...
-
-	return migrator.Up(ctx, db)
-}
-
-func registerMigrations(migrator *migrations.Migrator) {
-	migrator.RegisterMigrations(
-		migrations.Migration{
-			Up: func(tx bun.Tx) error {
-				if _, err := tx.Exec(`
+var _migrations = []migrations.Migration{
```
Comment (gfyrag) [inline ee/orchestration/internal/storage/migrations.go:10]: There, i moved migrations in an array.

File: ee/orchestration/internal/storage/migrations.go [hunk 2]
```diff
@@ -99,10 +91,40 @@ func registerMigrations(migrator *migrations.Migrator) {
...
+			return nil
+		},
+	},
+}
+
+func Migrate(ctx context.Context, db *bun.DB) error {
+	return MigrateUntil(ctx, db, len(_migrations))
+}
+
+func MigrateUntil(ctx context.Context, db *bun.DB, until int) error {
```
Comment (gfyrag) [inline ee/orchestration/internal/storage/migrations.go:125]: Add this function to be able to partially run migrations in tests.

File: ee/orchestration/internal/storage/migrations.go [hunk 3]
```diff
@@ -99,10 +91,40 @@ func registerMigrations(migrator *migrations.Migrator) {
...
-				return nil
-			},
+				return err
+			}
+			return nil
 		},
-	)
+	},
+	{
+		Up: func(tx bun.Tx) error {
```
Comment (gfyrag) [inline ee/orchestration/internal/storage/migrations.go:100]: The useful migration.

File: ee/orchestration/internal/workflow/config.go [hunk 4]
```diff
@@ -78,8 +78,7 @@ func (c *Config) run(ctx workflow.Context, db *bun.DB, instance Instance, variab
 	for ind, rawStage := range c.Stages {
 		logger.Info("run stage", "index", ind, "workflowID", instance.ID)
 
-		stage := NewStage(instance.ID, ind)
-
+		stage := NewStage(instance.ID, workflow.GetInfo(ctx).WorkflowExecution.RunID, ind)
```
Comment (gfyrag) [inline ee/orchestration/internal/workflow/config.go:81]: We add the temporal run id in the stage as compound primary key.

**Review by paul-nicolas — APPROVED** [review]

**Pull request merged**